src/app/signers/data/UseSigner.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  UseQueryResult,
+  useQueries,
+  useQueryClient,
+  useSuspenseQuery,
+} from '@tanstack/react-query';
+import { useCallback, useMemo, useState } from 'react';
+
+import { useSuspenseCurrentStackingCycle } from '../../../app/_components/Stats/CurrentStackingCycle/useCurrentStackingCycle';
+import { useGlobalContext } from '../../../common/context/useGlobalContext';
+import { TEN_MINUTES } from '../../../common/queries/query-stale-time';
+import {
+  SignerMetricsSignerForCycle,
+  useGetSignerMetricsSignerForCycleQuery,
+} from './signer-metrics-hooks';
+
+const SIGNER_QUERY_KEY = 'signer';
+
+export interface SignerInfo {
+  signing_key: string;
+  signer_address: string;
+  weight: number;
+  stacked_amount: string;
+  weight_percent: number;
+  stacked_amount_percent: number;
+  solo_stacker_count: number;
+  pooled_stacker_count: number;
+}
+
+export interface StackingHistoryInfo extends SignerMetricsSignerForCycle {
+  cycleid: number;
+}
+
+interface SignerError {
+  error: string;
+  message: string;
+  statusCode: number;
+}
+
+export function useSuspensePoxSigner(cycleId: number, signerKey: string) {
+  const { url: activeNetworkUrl } = useGlobalContext().activeNetwork;
+
+  return useSuspenseQuery<SignerInfo>({
+    queryKey: [SIGNER_QUERY_KEY, cycleId, signerKey],
+    queryFn: () =>
+      fetch(`${activeNetworkUrl}/extended/v2/pox/cycles/${cycleId}/signers/${signerKey}`).then(
+        res => res.json()
+      ),
+    staleTime: TEN_MINUTES,
+  });
+}
+
+const DEFAULT_LIST_LIMIT = 5;
+
+export function useSignerStackingHistory(signerKey: string, selectedCycle?: string) {
+  const { currentCycleId } = useSuspenseCurrentStackingCycle();
+  const getQuery = useGetSignerMetricsSignerForCycleQuery();
+  const [offset, setOffset] = useState(0);
+  const cyclesToQuery = useMemo(() => {
+    return selectedCycle
+      ? [parseInt(selectedCycle)]
+      : Array.from({ length: offset + DEFAULT_LIST_LIMIT }, (_, i) => currentCycleId - i);
+  }, [selectedCycle, currentCycleId, offset]);
+  const queryClient = useQueryClient();
+  const queries = useMemo(() => {
+    return cyclesToQuery.map(cycle => {
```

Comment:
It looks like this part will do multiple queries for all cycles? If that's the case I think it could cause problems with spawning many requests and spamming the API

Reply:
There isn't a batch endpoint, so I don't have any choice. Here is the endpoint that I am using to get signer data for a signer for a given cycle - https://signer-metrics-ck5uyle9e-hirosystems.vercel.app/#tag/Signers/operation/getPoxCycleSigner. As you can see I limit the queries to 5, which I think is an acceptable # of requests to make. Sure the user can spam this until there are no more cycles, but they could also do that on any of our other pages, and besides there aren't even 5 cycles to query for currently. 

TLDR: To resolve this we need a batch endpoint, but I don't think this is too bad, considering we've done similar and worse elsewhere